src/token.rs

```diff
@@ -778,6 +778,7 @@ macro_rules! Token {
     (,)           => { $crate::token::Comma };
     (/)           => { $crate::token::Div };
     (/=)          => { $crate::token::DivEq };
+    ($)           => { $crate::token::Dollar };
```

Comment:
This line does not compile on 1.15.0 which is our minimum supported Rust version. On old compilers we need to not have this rule.
https://travis-ci.org/dtolnay/syn/jobs/550313978

Reply:
I'm not sure what the best way is to make this conditional on the compiler version. (I'm also not sure if that is what you're suggesting here.) Also I personally would find it confusing if this feature is dependent on the compiler version used. It'd be easy for code using syn to accidentally depend on a newer rustc version by using `Token![$]`.

Should I open another PR instead that simply documents that `Token![$]` isn't supported and why?

Reply:
I think having it conditional on the compiler version would be okay. We have some version checks in [syn/build.rs](https://github.com/dtolnay/syn/blob/0.15.38/build.rs) already. You can introduce a cfg that is defined on new enough compilers that are able to support Token![$]. It's possible this will require duplicating the whole Token macro definition but that's fine.

Downstream code accidentally depending on new rustc versions is not a thing we need to consider because regardless of what happens in Syn it's always easy to accidentally depend on new standard library APIs or language features. Crates that care will have a CI setup that catches this (like Syn does).